synapse/storage/data_stores/main/room.py

```diff
@@ -349,9 +401,13 @@ def is_room_blocked(self, room_id):
         # for, and another query for getting the total number of events that could be
         # returned. Thus allowing us to see if there are more events to paginate through
         info_sql = """
-            SELECT state.room_id, state.name, state.canonical_alias, curr.joined_members
+            SELECT state.room_id, state.name, state.canonical_alias, state.encryption,
+              state.guest_access, state.is_federatable, state.history_visibility,
+              state.join_rules, rooms.room_version, rooms.is_public, rooms.creator,
+              curr.joined_members, curr.current_state_events, curr.local_users_in_room
             FROM room_stats_state state
             INNER JOIN room_stats_current curr USING (room_id)
+            INNER JOIN rooms USING (room_id)
```

Comment:
Any idea if there are performance impacts to joining `rooms` here?

Reply:
I do not know it. I am not a SQL specialist.

Reply:
`explain` for the previous version with some example values:

```
synapse=# explain SELECT state.room_id, state.name, state.canonical_alias, curr.joined_members
synapse-# FROM room_stats_state state
synapse-# INNER JOIN room_stats_current curr USING (room_id)
synapse-# ORDER BY state.name
synapse-# LIMIT 200
synapse-# OFFSET 5;
                                             QUERY PLAN                                             
----------------------------------------------------------------------------------------------------
 Limit  (cost=255.31..255.81 rows=200 width=90)
   ->  Sort  (cost=255.30..260.44 rows=2058 width=90)
         Sort Key: state.name
         ->  Hash Join  (cost=76.99..165.99 rows=2058 width=90)
               Hash Cond: (state.room_id = curr.room_id)
               ->  Seq Scan on room_stats_state state  (cost=0.00..83.58 rows=2058 width=86)
               ->  Hash  (cost=50.33..50.33 rows=2133 width=36)
                     ->  Seq Scan on room_stats_current curr  (cost=0.00..50.33 rows=2133 width=36)
(8 rows)

synapse=# 
```

and with the new version:

```
synapse=# explain SELECT state.room_id, state.name, state.canonical_alias, curr.joined_members,
synapse-# curr.local_users_in_room, rooms.room_version, rooms.creator,
synapse-# state.encryption, state.is_federatable, rooms.is_public, state.join_rules,
synapse-# state.guest_access, state.history_visibility, curr.current_state_events
synapse-# FROM room_stats_state state
synapse-# INNER JOIN room_stats_current curr USING (room_id)
synapse-# INNER JOIN rooms USING (room_id)
synapse-# ORDER BY rooms.room_version
synapse-# LIMIT 200
synapse-# OFFSET 5;
                                             QUERY PLAN                                             
----------------------------------------------------------------------------------------------------
 Limit  (cost=331.68..332.18 rows=200 width=162)
   ->  Sort  (cost=331.67..336.44 rows=1907 width=162)
         Sort Key: rooms.room_version
         ->  Hash Join  (cost=154.90..248.91 rows=1907 width=162)
               Hash Cond: (state.room_id = curr.room_id)
               ->  Hash Join  (cost=77.91..166.90 rows=1907 width=182)
                     Hash Cond: (state.room_id = rooms.room_id)
                     ->  Seq Scan on room_stats_state state  (cost=0.00..83.58 rows=2058 width=131)
                     ->  Hash  (cost=54.07..54.07 rows=1907 width=51)
                           ->  Seq Scan on rooms  (cost=0.00..54.07 rows=1907 width=51)
               ->  Hash  (cost=50.33..50.33 rows=2133 width=44)
                     ->  Seq Scan on room_stats_current curr  (cost=0.00..50.33 rows=2133 width=44)
(12 rows)

synapse=# 
```

So a ~30% increase on the upper bound, but this query is currently only run by the admin API, and as we're getting much more information as a result of it, I don't think this is too much of an issue.